src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -911,29 +907,80 @@ func (r *ClusterReconciler) setInitialSuperUserPassword(
 		return nil
 	}
 
+	// list out users here
+	users, err := adminAPI.ListUsers(ctx)
+	if err != nil {
+		if !strings.Contains(strings.ToLower(err.Error()), "unavailable") {
+			return fmt.Errorf("could not fetch users: %w", err)
+		}
+	}
+
+	errs := make([]error, 0)
 	for _, obj := range objs {
-		var secret corev1.Secret
-		err := r.Get(ctx, types.NamespacedName{
-			Namespace: obj.Namespace,
-			Name:      obj.Name,
-		}, &secret)
+		// We first check that the secret has been created
+		// This should have been done by this point, if not
+		// requeue, this is created by the controller and is
+		// the source of truth, not the admin API
+		var secret *corev1.Secret
+		err := r.Get(ctx, obj, secret)
 		if err != nil {
-			return fmt.Errorf("fetching Secret (%s) from namespace (%s): %w", obj.Name, obj.Namespace, err)
+			errs = append(errs, fmt.Errorf("could not fetch user secret (%s): %w", obj.String(), err))
+			continue
 		}
+		// if we do not find the user in the list, we should create them
+		userFound := false
+		expectedUser := string(secret.Data[corev1.BasicAuthUsernameKey])
 
-		username := string(secret.Data[corev1.BasicAuthUsernameKey])
-		password := string(secret.Data[corev1.BasicAuthPasswordKey])
+		for i := range users {
+			if users[i] == expectedUser {
+				userFound = true
+				break
+			}
+		}
 
-		err = adminAPI.CreateUser(ctx, username, password, admin.ScramSha256)
-		// {"message": "Creating user: User already exists", "code": 400}
-		if err != nil && !strings.Contains(err.Error(), "already exists") { // TODO if user already exists, we only receive "400". Check for specific error code when available.
-			return &resources.RequeueAfterError{
-				RequeueAfter: resources.RequeueDuration,
-				Msg:          fmt.Sprintf("could not create user: %v", err),
+		if userFound {
+			// update the user here, we cannot retrieve password changes here to date
+			if err := updateUserOnAdminAPI(ctx, adminAPI, secret); err != nil {
+				errs = append(errs, fmt.Errorf("could not update user (%s): %w", obj.String(), err))
```

Comment:
I'm thinking we be a little more verbose, here, so a user that doesn't know how it works can have some idea which logs to look for.

```suggestion
				errs = append(errs, fmt.Errorf("could not update the user (%s) in the Redpanda admin api: %w", obj.String(), err))
```